udata/core/dataset/views.py

```diff
@@ -86,8 +88,86 @@ def get_context(self):
         context['can_edit'] = DatasetEditPermission(self.dataset)
         context['can_edit_resource'] = ResourceEditPermission
         context['discussions'] = Discussion.objects(subject=self.dataset)
+
         return context
 
+    def get_json_ld(self):
+        dataset = self.dataset
+
+        result = {"@context": "http://schema.org",
+                  "@type": "Dataset",
+                  "@id": str(dataset.id),
+                  "description": dataset.description,
+                  "alternateName": dataset.slug,
+                  "dateCreated": dataset.created_at.isoformat(),
+                  "dateModified": dataset.last_modified.isoformat(),
+                  "url": url_for('datasets.show', dataset=dataset, _external=True),
+                  "name": dataset.title,
+                  "keywords": ', '.join(dataset.tags),
+                  "distribution": map(self.get_json_ld_resource, dataset.resources),
+                  # This value is not standard
+                  "extras": map(self.get_json_ld_extra, dataset.extras.items())
+        }
+
+        if dataset.license and dataset.license.url:
+            result["license"] = dataset.license.url
+
+        if dataset.organization:
+            view = OrganizationDetailView()
```

Comment:
Now, I'm sure it can't.
If the model have the method, you could directly write:
```python
if dataset.organization:
    author = dataset.organization.__jsonld__
```
This is an example with models having:
```python
@property  # even better with @cachedproperty
def __jsonld__(self):
    return {
        #JSONLD
    }
```
This is simply a proposal.